### DISCUSSION TITLE
Segmentation Fault related to MinPackMover being called as part of an IteratedConvergence Mover

### DISCUSSION THREAD

**Discussion by tsamdin** [RosettaCommons/rosetta#261] [Rosetta]
I'm not sure how to troubleshoot this segmentation fault. I know that it occurs when an IteratedConvergence Mover, which calls on a MinPackMover is started, but not much else. 

It's not obvious to me what the ROSETTA_CRASH.log is pointing towards as an issue. 

Attached below is the relevant section of my XML script: 

MinPackMover name="MPM" scorefxn="r12_cart" nonideal="1" off_rotamer_pack="1" cartesian="1"
        task_operations="use_input_rotamer,extrarot,no_repack_L_protein,no_design_L_protein,L_only_design,D_only_design,D_buried,L_buried,D_strand_buried_resnum,D_strand_exposed_resnum,no_CYS,no_ASP,no_MET,aromachi,no_GLY" 
        
IteratedConvergence name="converge" mover="MPM" filter="fa_rep_sol" delta="20.0" cycles="5" maxcycles="25"

and here's part of the output:

core.energy_methods.AACompositionEnergy: Cleaning up cached AACompositionEnergy data after packing.
protocols.rosetta_scripts.ParsedProtocol: setting status to success
protocols.rosetta_scripts.ParsedProtocol: =======================BEGIN MOVER IteratedConvergence - converge=======================
core.energy_methods.CartesianBondedEnergy: Creating new peptide-bonded energy container (56)

AN INTERNAL ERROR HAS OCCURED. PLEASE SEE THE CONTENTS OF ROSETTA_CRASH.log FOR DETAILS.

/var/spool/slurm/slurmd/job42297788/slurm_script: line 9: 1330569 Segmentation fault      (core dumped) /data/samdintd/rosetta/rosetta.source.release-371/main/source/bin/rosetta_scripts.default.linuxgccrelease -in:file:s relax/2GB1_S1_C.pdb -in:file:native native/2GB1_native.pdb -parser:protocol xml/2GB1_ripple_Std1_1.xml @inputs/rosetta_200.flags -out:prefix output_2GB1_S1_1/

And the Rosetta Crash log:**

[START_CRASH_REPORT]
[ROSETTA_VERSION]: 2024.09+release.06b3cf8
[COMMIT_DATE]: 2024-03-07T14:59:22.624665
[APPLICATION]: /data/samdintd/rosetta/rosetta.source.release-371/main/source/bin/rosetta_scripts.default.linuxgccrelease
[MODE]: Release
[EXTRAS]: default
[OS]: GNU/Linux
[COMPILER]: GCC version "8.5.0 20210514 (Red Hat 8.5.0-22)"
[STDLIB]: libstdc++ version 20210514
[START_OPTIONS]
 -in:file:s=relax/2GB1_S1_C.pdb -in:file:native=native/2GB1_native.pdb -in:file:fullatom -inout:write_all_connect_info -out:nstruct=200 -out:prefix=output_2GB1_S1_1/ -run:preserve_header -parser:protocol=xml/2GB1_ripple_Std1_1.xml -jd2:failed_job_exception=false -score:symmetric_gly_tables

[END_OPTIONS]

[START_BACKTRACE]: RAW_LIBC
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libutility.so(utility::save_crash_report(char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x23) [0x155543fe07f3]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libutility.so(utility::signal_handler(int)+0xd4) [0x155543fe0904]
/lib64/libc.so.6(+0x4eb50) [0x15554099db50]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libcore.5.so(core::pack::scmin::ResidueAtomTreeCollection::set_active_restype_index(unsigned long)+0x24) [0x1555494ca4c4]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libcore.5.so(core::pack::scmin::ResidueAtomTreeCollection::update_from_momento(core::pack::scmin::ResidueAtomTreeCollectionMomento const&)+0x23) [0x1555494cb273]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libcore.5.so(core::pack::off_rotamer_pack_update_pose(core::pose::Pose&, core::pack::rotamer_set::ContinuousRotamerSets const&, std::shared_ptr<core::pack::scmin::AtomTreeCollection>, core::pack::scmin::SidechainStateAssignment const&)+0x5d) [0x155548e6d66d]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libcore.5.so(core::pack::off_rotamer_pack(core::pose::Pose&, core::scoring::ScoreFunction const&, std::shared_ptr<core::pack::task::PackerTask const>)+0x190) [0x155548e75220]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.3.so(protocols::minimization_packing::MinPackMover::apply(core::pose::Pose&)+0x84) [0x15554c576894]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::moves::IteratedConvergenceMover::apply(core::pose::Pose&)+0xf6) [0x15554a82bff6]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::rosetta_scripts::ParsedProtocol::apply_mover(core::pose::Pose&, protocols::rosetta_scripts::ParsedProtocol::ParsedProtocolStep const&)+0x211) [0x15554a960e31]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::rosetta_scripts::ParsedProtocol::apply_step(core::pose::Pose&, protocols::rosetta_scripts::ParsedProtocol::ParsedProtocolStep const&, bool)+0x1a) [0x15554a96123a]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::rosetta_scripts::ParsedProtocol::sequence_protocol(core::pose::Pose&, __gnu_cxx::__normal_iterator<protocols::rosetta_scripts::ParsedProtocol::ParsedProtocolStep const*, std::vector<protocols::rosetta_scripts::ParsedProtocol::ParsedProtocolStep, std::allocator<protocols::rosetta_scripts::ParsedProtocol::ParsedProtocolStep> > >)+0x40) [0x15554a9618f0]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::rosetta_scripts::ParsedProtocol::apply(core::pose::Pose&)+0x2a6) [0x15554a962576]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::jd2::JobDistributor::run_one_job(std::shared_ptr<protocols::moves::Mover>&, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, unsigned long&, unsigned long&, bool)+0x107f) [0x15554a4c72df]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::jd2::JobDistributor::go_main(std::shared_ptr<protocols::moves::Mover>)+0x14e) [0x15554a4c86ee]
/gpfs/gsfs12/users/samdintd/rosetta/rosetta.source.release-371/main/source/build/src/release/linux/4.18/64/x86/gcc/8/default/libprotocols.1.so(protocols::jd2::FileSystemJobDistributor::go(std::shared_ptr<protocols::moves::Mover>)+0x47) [0x15554a4b1007]
/data/samdintd/rosetta/rosetta.source.release-371/main/source/bin/rosetta_scripts.default.linuxgccrelease() [0x404457]
/lib64/libc.so.6(__libc_start_main+0xe5) [0x155540989d85]
/data/samdintd/rosetta/rosetta.source.release-371/main/source/bin/rosetta_scripts.default.linuxgccrelease() [0x40468e]

[END_BACKTRACE]

[FILE]: SIGSEGV
[LINE]: 11
[START_MESSAGE]
Segmentation Fault

[END_MESSAGE]
[END_CRASH_REPORT]

**Comment by roccomoretti** [comment]
Segmentation faults _per se_ are hard to debug. I'm not seeing anything from the tracer output or the crash log which is pointing me to an explanation.

If at all possible, (re)compiling in debug mode and then running that would help. Debug mode adds in additional checks which help to diagnose issues (but which slow the running down, which is why they're off in release mode). Hopefully those would point to what might be going on.
